test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexMatcherTest.scala

```diff
@@ -285,23 +285,49 @@ class RegexMatcherTest  {
     assertSame(matcher0, matcher0.region(0, 3))
     assertEquals(0, matcher0.regionStart)
     assertEquals(3, matcher0.regionEnd)
+    assertFalse(matcher0.lookingAt())
+    matcher0.region(0, 3)
+    assertFalse(matcher0.matches())
+    matcher0.region(0, 3)
     assertFalse(matcher0.find())
 
     matcher0.region(0, 15)
     assertEquals(0, matcher0.regionStart)
     assertEquals(15, matcher0.regionEnd)
+    assertFalse(matcher0.lookingAt())
+    matcher0.region(0, 15)
+    assertFalse(matcher0.matches())
+    matcher0.region(0, 15)
     assertTrue(matcher0.find())
     assertEquals("Scalable", matcher0.group)
 
     matcher0.region(2, 7)
     assertEquals(2, matcher0.regionStart)
     assertEquals(7, matcher0.regionEnd)
+    assertTrue(matcher0.lookingAt())
+    matcher0.region(2, 7)
```

Comment:
Do we need to / want to reset the region here? The code suggests we don't need to.

Reply:
Oh, I see. This is probably affected by #4254...

Reply:
Yes, we need to. The code suggests we don't need to because it's broken (#4254), but if we don't reset the region, the test does not pass on the JVM.